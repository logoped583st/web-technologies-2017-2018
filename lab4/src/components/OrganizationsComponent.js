import React from 'react';
import { connect } from "react-redux";


function BioComponent(props) {
    console.log(props.login)
    return (
        props.login !== '' && <div className="ORGS">

            <h2>Organizations</h2>
            <div className="orgs_adds">
                <h3>{props.login}</h3>
                <img src={props.image} alt={props.image} />
            </div>
        </div>
    );
}

const getState = (state) => {
    return {
        login: state.reducerOrgs.login,
        image: state.reducerOrgs.image
    };
};

export default connect(getState)(BioComponent);