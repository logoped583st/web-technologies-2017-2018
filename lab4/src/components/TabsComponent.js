import React from 'react';
import { connect } from "react-redux";
import { Tab, Tabs, TabList, TabPanel } from 'react-tabs';

class TabsComponent extends React.Component {
    constructor() {
        super();
        this.state = { tabIndex: 0 };
    }

    render() {
        return (

            <Tabs selectedIndex={this.state.tabIndex} onSelect={tabIndex => this.setState({ tabIndex })}>
                <TabList>
                    <Tab>Основное</Tab>
                    <Tab>Образование</Tab>
                    <Tab>Контакты</Tab>
                </TabList>
                <TabPanel><button >Edit1</button></TabPanel>
                <TabPanel><button >Edit2</button></TabPanel>
                <TabPanel><button >Edit3</button></TabPanel>
            </Tabs>
        );
    }


}



export default connect(null)(TabsComponent);