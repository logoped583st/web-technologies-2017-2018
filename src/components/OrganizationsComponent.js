import React from 'react';
import { connect } from "react-redux";


function OrganizationsComponent(props) {
    var OrgsArr = [];

    for (const org in props.Orgs) {
        OrgsArr.push(props.Orgs[org])
    }

    return (
        (OrgsArr.length !== 0) && <div className="ORGS">

            <h2>Organizations</h2>

            {OrgsArr.map(function (item) {
                return (<div className="orgs_adds">
                    <h3>{item.login}</h3>
                    <img src={item.avatar_url} alt={item.avatar_url} />
                </div>)
            })}


        </div>
    );
}

const getState = (state) => {
    return {
        Orgs: state.reducerOrgs,
    };
};

export default connect(getState)(OrganizationsComponent);