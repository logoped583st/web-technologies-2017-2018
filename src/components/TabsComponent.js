import React from 'react';
import { Tab, Tabs, TabList, TabPanel } from 'react-tabs';
import TextAreaComponent from './TextAreaComponent';

class TabsComponent extends React.Component {

    constructor() {
        super();
        this.state = { tabIndex: 0 };
    }

    render() {
        return (

            <Tabs selectedIndex={this.state.tabIndex} onSelect={tabIndex => this.setState({ tabIndex })}>
                <TabList>
                    <Tab>Основное</Tab>
                    <Tab>Образование</Tab>
                    <Tab>Контакты</Tab>
                </TabList>
                <TabPanel>
                    <TextAreaComponent id="textArea1" number={1} />
                </TabPanel>
                <TabPanel>
                    <TextAreaComponent id="textArea2" number={2} />
                </TabPanel>
                <TabPanel>
                    <TextAreaComponent id="textArea3" number={3} />
                </TabPanel>
            </Tabs>
        );
    }


}



export default (TabsComponent);