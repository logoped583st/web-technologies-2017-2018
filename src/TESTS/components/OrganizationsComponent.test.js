import React from 'react';
import OrganizationsComponent from '../../components/OrganizationsComponent';
import TabsComponent from '../../components/TabsComponent'
import { shallow, mount, configure } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

configure({ adapter: new Adapter() });

describe('Organization component test', () => {
    const mockStore = configureStore([ thunk ]);
    const initialState = {
        reducerUser: {
            login: 'logoped583st'
        },
        reducerOrgs: [
            {
                avatar_url: 'https://avatars3.githubusercontent.com/u/30079690?s=400&u=e3443b749fa4d36979e0d631192f3a7bbadc3eeb&v=4',
                login: 'logoped583st'
            }
        ]
    }

    let store;

    beforeEach(() => {
        store = mockStore(initialState);
    })

    it('MainUserInfoComponent component rendered succesfully', () => {
        
        const reducerOrgs = mount(<OrganizationsComponent store={store} />);
        expect(mountToJson(reducerOrgs)).toMatchSnapshot();
    });

    it('User props match store', () => {
        const reducerOrgs = shallow(<OrganizationsComponent store={store}/>);
        expect(reducerOrgs.prop('avatar_url')).toBe(undefined);
        expect(reducerOrgs.prop('login')).toBe(undefined);
    })
});