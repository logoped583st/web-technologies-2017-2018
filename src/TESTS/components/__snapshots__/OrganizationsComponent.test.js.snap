// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organization component test MainUserInfoComponent component rendered succesfully 1`] = `
<Connect(OrganizationsComponent)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <OrganizationsComponent
    Orgs={
      Array [
        Object {
          "avatar_url": "https://avatars3.githubusercontent.com/u/30079690?s=400&u=e3443b749fa4d36979e0d631192f3a7bbadc3eeb&v=4",
          "login": "logoped583st",
        },
      ]
    }
    dispatch={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <div
      className="ORGS"
    >
      <h2>
        Organizations
      </h2>
      <div
        className="orgs_adds"
      >
        <h3>
          logoped583st
        </h3>
        <img
          alt="https://avatars3.githubusercontent.com/u/30079690?s=400&u=e3443b749fa4d36979e0d631192f3a7bbadc3eeb&v=4"
          src="https://avatars3.githubusercontent.com/u/30079690?s=400&u=e3443b749fa4d36979e0d631192f3a7bbadc3eeb&v=4"
        />
      </div>
    </div>
  </OrganizationsComponent>
</Connect(OrganizationsComponent)>
`;
